### DISCUSSION TITLE
Remove tweens used in scroll_to() function

### DISCUSSION THREAD

**Discussion by HaroldLever** [SpyrexDE/SmoothScroll#58] [Ideas]
I happened to find that `scroll_horizontal` and `scroll_vertical` didn't kill tweens used in scroll_to() function properly. I was intending to create a new pull request, then I realized maybe we shouldn't use tweens, but should apply the exact velocity to make content scroll to destination. Future developers will no longer need to think about when to kill tweens. 

I have some worries if we apply a velocity instead of using tween. When `wheel_scroll_damper` and `dragging_scroll_damper` are different, scroll_to() behavior will present differently. Also, scroll_to() will remove the parameter `duration` since we can't control it, which seems to be a "downgrade" and more importantly, it's not a compatible change.

I want to hear your idea on this topic, which solution should we take and is there any better thoughts?

**Comment by SpyrexDE** [comment]
Generally, I agree that moving from tweens to adding velocity is a good idea.

> When wheel_scroll_damper and dragging_scroll_damper are different, scroll_to() behavior will present differently.

For that, we could add a parameter with default value where the user can specify which `scroll_damper` to use.

> Also, scroll_to() will remove the parameter duration since we can't control it, which seems to be a "downgrade" and more importantly, it's not a compatible change.

I think this drawback is worth it. If there is demand on bringing that `duration` parameter back in we can maybe add it back later using custom calculations and states.
